src/ServiceControl.TransportAdapter/ServiceControlTransportAdapter.cs

```diff
@@ -10,15 +10,21 @@ class ServiceControlTransportAdapter<TEndpoint, TServiceControl> : ITransportAda
         internal ServiceControlTransportAdapter(
             FailedMessageForwarder<TEndpoint, TServiceControl> failedMessageForwarder,
             ControlForwarder<TEndpoint, TServiceControl> controlMessageForwarder,
-            AuditForwarder<TEndpoint, TServiceControl> auditForwarder)
+            AuditForwarder<TEndpoint, TServiceControl> auditForwarder,
+            NServiceBusMetricReport<TServiceControl> report)
         {
             this.failedMessageForwarder = failedMessageForwarder;
             this.controlMessageForwarder = controlMessageForwarder;
             this.auditForwarder = auditForwarder;
+            this.report = report;
         }
 
         public async Task Start()
         {
+            if (report != null)
```

Comment:
No particular reason other than "what starts last should be stopped early". All three forwarders could be started concurrently because they are independent. In fact I'll change this to do it like that. Regarding the reporter, it makes sense to start it first and stop it last.